libs/isograph-react/src/loadable-hooks/useConnectionSpecPagination.ts

```diff
@@ -92,7 +106,9 @@ export function useConnectionSpecPagination<
   TItem,
 >(
   loadableField: LoadableField<TReadFromStore, Connection<TItem>>,
+  ...[initialArgs]: UseConnectionSpecArgs<TReadFromStore['parameters']>
 ): UsePaginationReturnValue<TReadFromStore, TItem> {
+  const [args] = useState(initialArgs);
```

Comment:
If we pass extra arguments, like filters, then we need them for next pages

Reply:
Three thoughts:

- Good point, we certainly don't correctly handle other params at the moment. Maybe we punt for now on additional args, and just have you pass additional args via the isograph literal, since we should be recreating everything if the args change, and that requires that `change -> uncommited state` hook.
- Right now, if you pass different variables via the isograph literal, all the state gets recreated, so maybe that is sufficient for now. (I haven't thought this through all the way.)
- I think that this hook should take three params, the loadable field, the initial state, and the additional args that are passed to each `fetchMore`

Reply:
This makes sens to me too, I thought if we go this way, then user will have to add `key={}` to reset components.
Great thing you've noticed it will recreating everything.